helix-core/src/main/java/org/apache/helix/manager/zk/GenericBaseDataAccessorBuilder.java

```diff
@@ -83,7 +82,7 @@ protected RealmAwareZkClient createZkClient(RealmAwareZkClient.RealmMode realmMo
       case MULTI_REALM:
         try {
           zkClient = new FederatedZkClient(connectionConfig, clientConfig);
-        } catch (IOException | InvalidRoutingDataException e) {
+        } catch (InvalidRoutingDataException e) {
```

Comment:
It was removed from the signature. We don't throw it anymore. It will be thrown in the form of a MultiZkException, and will be a RuntimeException, not a typed exception.